src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -35,7 +35,7 @@ import com.databricks.spark.xml.XmlOptions
 /**
  * Wraps parser to iteration process.
  */
-private[xml] object StaxXmlParser {
+private[xml] object StaxXmlParser extends Serializable {
```

Comment:
Not clear why, but this caused a bunch of lambdas to not serialize in 2.12. It should be harmless to let the object serialize; it has no fields.